Support schildichat:// links

AndrewRyanChama opened the pull request:
We want to suppot schildichat:// links in addition to element:// links.

This change adds support for schildichat:// links of the same format as
the element links.

tested on a phone with android 11 and google chrome
schildichat://room/#web:schildi.chat

full disclosure this doesn't address https://github.com/vector-im/element-android/issues/5429 at all and there might be merge conflicts if that ever gets fixed (likely never)

SpiritCroc commented on vector/src/main/AndroidManifest.xml:
```diff
@@ -140,6 +140,7 @@
                 <data
                     android:host="connect"
                     android:scheme="element" />
+                <data android:scheme="schildichat" />
```
We'd rather not touch that one, as I don't care enough about SSO to find proper ways to test it. Also, you didn't touch the files handling SSO connect things, so this might not work either way.

SpiritCroc replied:
We'd rather not touch that one, as I don't care enough about SSO to find proper ways to test it. Also, you didn't touch the files handling SSO connect things, so this might not work either way.

SpiritCroc commented on vector/src/main/AndroidManifest.xml:
```diff
@@ -242,7 +243,7 @@
                     android:host="room"
                     android:scheme="element" />
-
+                <data android:scheme="schildichat" />
```
Why not restrict to `room` and `user` like upstream does? Furthermore, this likely conflicts with the one above since you haven't restricted the SSO one either?

SpiritCroc commented on vector/src/main/java/im/vector/app/features/home/HomeActivity.kt:
```diff
@@ -274,7 +277,7 @@ class HomeActivity :
             val resolvedLink = when {
                 // Element custom scheme is not handled by the sdk, convert it to matrix.to link for compatibility
-                deepLink.startsWith(MATRIX_TO_CUSTOM_SCHEME_URL_BASE) -> {
+                deepLink.startsWith(MATRIX_TO_CUSTOM_SCHEME_URL_BASE)    -> {
```
Rather not touch indention here: we care more about having less merge conflicts with upstream, then proper code style

SpiritCroc replied:
Rather not touch indention here: we care more about having less merge conflicts with upstream, than proper code style. That comment also applies to more changes, I'll not comment on each of them

SpiritCroc commented on vector/src/main/java/im/vector/app/features/permalink/PermalinkHandler.kt:
```diff
@@ -220,6 +220,9 @@ class PermalinkHandler @Inject constructor(private val activeSessionHolder: Acti
         const val MATRIX_TO_CUSTOM_SCHEME_URL_BASE = "element://"
         const val ROOM_LINK_PREFIX = "${MATRIX_TO_CUSTOM_SCHEME_URL_BASE}room/"
         const val USER_LINK_PREFIX = "${MATRIX_TO_CUSTOM_SCHEME_URL_BASE}user/"
+        const val SCHILDI_CHAT_CUSTOM_SCHEME_URL_BASE = "schildichat://"
```
Let's use the upstream names for these variables with prepended "SC_":
- SC_MATRIX_TO_CUSTOM_SCHEME_URL_BASE
- SC_ROOM_LINK_PREFIX
- SC_USER_LINK_PREFIX

SpiritCroc commented on vector/src/main/AndroidManifest.xml:
```diff
@@ -242,7 +243,7 @@
                     android:host="room"
                     android:scheme="element" />
-
+                <data android:scheme="schildichat" />
```
Why not restrict to `room` and `user` like upstream does? Furthermore, this likely conflicts with the one above since you haven't restricted the SSO one either?

AndrewRyanChama replied:
All of the ```<data>``` tags are merged together, so the ```android:host``` attributes from above apply also to schildichat.  I can add the same ```android:host="room"``` tags as above if you want but they would have no effect and I want to keep fewer lines changed here.

see:
https://developer.android.com/training/app-links/deep-linking#adding-filters

SpiritCroc replied:
Ah, good to know! Then that's fine

SpiritCroc reviewed (changes requested):
Thanks for this PR, looks good for the most part! Haven't tested yet, but giving a few comments on the code

AndrewRyanChama committed:
Support schildichat:// links

We want to suppot schildichat:// links in addition to element:// links.

This change adds support for schildichat:// links of the same format as
the element links.
